File: PHP/problem_3/algorithm.php
<?php

function maxPrimeFactors($n) {
    $maxPrime = -1;

    while ($n % 2 == 0) {
        $maxPrime = 2;
        $n /= 2;
    }

    for ($i = 3; $i <= sqrt($n); $i += 2) {
        while ($n % $i == 0) {
            $maxPrime = $i;
            $n /= $i;
        }
    }

    if ($n > 2) {
        $maxPrime = $n;
    }

    return $maxPrime;
}

function main() {
    $n = maxPrimeFactors(600851475143);
    echo $n . "\n";
}

main();

File: Java/problem_3/Algorithm.java
import java.lang.Math;

class Algorithm {

    public static long maxPrimeFactors(long n) {
      long maxPrime = -1;

      while (n % 2 == 0) {
          maxPrime = 2;
          n /= 2;
      }

      for (int i = 3; i <= Math.sqrt(n); i += 2) {
          while (n % i == 0) {
              maxPrime = i;
              n /= i;
          }
      }

      if (n > 2) {
          maxPrime = n;
      }

      return maxPrime;
    }

    public static void main(String[] args) {
        long n = maxPrimeFactors(600851475143L);

        System.out.println(n);
    }

}
